datasafe-business/src/test/java/de/adorsys/datasafe/business/impl/e2e/BasicFunctionalityWithConcurrencyTest.java

```diff
@@ -99,89 +116,44 @@ void writeToPrivateListPrivateInDifferentThreads(WithStorageProvider.StorageDesc
-
-        validateUserPrivateStorage(testPath, privateJohnFiles, expectedData, john);
-
-        validateUserPrivateStorage(testPath, privateJaneFiles, expectedData, jane);
 
-    }
-
-    private void init(WithStorageProvider.StorageDescriptor descriptor) {
-        DefaultDatasafeServices datasafeServices = DatasafeServicesProvider
-                .defaultDatasafeServices(descriptor.getStorageService(), descriptor.getLocation());
-        initialize(datasafeServices);
-
-        this.location = descriptor.getLocation();
-        this.storage = descriptor.getStorageService();
+        metricCollector.setDataSize(size);
+        metricCollector.setStorageType(storage.getClass().getSimpleName());
+        metricCollector.setNumberOfThreads(poolSize);
+        metricCollector.writeToJSON();//json files in target folder
     }
 
     private void createFileForUserParallelly(ThreadPoolExecutor executor, CountDownLatch holdingLatch,
-                                             CountDownLatch finishHoldingLatch, byte[] bytes,
-                                             UserIDAuth john) {
+                                             CountDownLatch finishHoldingLatch, String testFilePath,
+                                             UserIDAuth user) {
         AtomicInteger counter = new AtomicInteger();
         String remotePath = "folder2";
 
         for (int j = 0; j < NUMBER_OF_TEST_FILES; j++) {
             executor.execute(() -> {
                 try {
-                    holdingLatch.await(TIMEOUT_S, TimeUnit.SECONDS);
+                    holdingLatch.await();
 
-                    long measurementOfWritingTextToFile = System.currentTimeMillis();
-                    Thread.currentThread().setName(john.getUserID().getValue());
+                    Thread.currentThread().setName(user.getUserID().getValue());
 
                     String filePath = remotePath + "/" + counter.incrementAndGet() + ".txt";
 
                     log.debug("Saving file: {}", filePath);
-                    writeDataToFileForUser(john, filePath, bytes, finishHoldingLatch);
 
-                    loadReport.add(Thread.currentThread().getName() + " write data to user " + john.getUserID().getValue() + ": " +
-                            (System.currentTimeMillis() - measurementOfWritingTextToFile) + "ms");
+                    Instant startSaving = Instant.now();
+
+                    writeDataToFileForUser(user, filePath, testFilePath, finishHoldingLatch);
+
+                    long durationOfSavingFile = Duration.between(startSaving, Instant.now()).toMillis();
+                    metricCollector.addSaveRecord(
+                           user.getUserID().getValue(),
+                           SaveTestRecord
+                                   .builder()
+                                   .duration(durationOfSavingFile)
+                                   .threadName(Thread.currentThread().getName())
```

Comment:
Given user is supplied here, it might be better to have SaveTestRecord binded to user id instead of thread name, thread name equal to user id is nice feature for logs, not for data transfer